perf: Materialize elements_chain

timgl opened the pull request:
## Problem

elements_chain is a huge column, and we use regex to parse it, so queries are slow. I looked at [the stats](https://metabase.prod-us.posthog.dev/question/799-action-autocapture-stats), and about half of actions have either text or href filters, so we can just extract those and query them directly, instead of through parsing elements_chain.

Of actions that only have selected, about half have matching on ID. Another 25% match on one of the 

<!-- Who are we building for, what are their needs, why is this important? -->

## Changes

Add 4 materialized columns. I'll add the logic to actually query this later, that way we won't get "column doesn't exist" type issues.

<!-- If there are frontend changes, please include screenshots. -->
<!-- If a reference design was involved, include a link to the relevant Figma frame! -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

<!-- Yes / no / it doesn't have an impact. -->

## How did you test this code?

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

timgl committed:
perf: Materialize elements_chain

pauldambra committed:
chore: add even more replay ingestion metrics to the hourly task (#23169)

timgl committed:
fix

github-actions[bot] committed:
Update query snapshots

pauldambra committed:
chore: log error if need be (#23171)

fuziontech commented on posthog/clickhouse/migrations/0064_materialize_elements_chain.py:
```diff
@@ -0,0 +1,37 @@
+
+def add_columns_to_required_tables(_):
+    with ch_pool.get_client() as client:
+        client.execute(ADD_COLUMNS_BASE_SQL.format(table="events", cluster=CLICKHOUSE_CLUSTER))
+
+        client.execute(ADD_COLUMNS_BASE_SQL.format(table="writable_events", cluster=CLICKHOUSE_CLUSTER))
```
Even though you technically _can_ create `MATERIALIZED` columns on a distributed table like this now, this goes against how we do materialization. I'd take a look at `events` and `sharded_events` to see the difference, but the gist is you only create the materialized column on the `sharded` (actual MergeTree) table and then read through through that on the `events` distributed table. **You _do not_ want to put _any_ materialized columns or references to materialized columns on the writable table. This can break ingestion.**

You can see that on the `writable_events` table as it is on prod right now:
```sql
CREATE TABLE posthog.writable_events
(
    `uuid` UUID,
    `event` String,
    `properties` String,
    `timestamp` DateTime64(6, 'UTC'),
    `team_id` Int64,
    `distinct_id` String,
    `elements_hash` String,
    `created_at` DateTime64(6, 'UTC'),
    `_timestamp` DateTime,
    `_offset` UInt64,
    `elements_chain` String,
    `person_id` UUID,
    `person_properties` String,
    `group0_properties` String,
    `group1_properties` String,
    `group2_properties` String,
    `group3_properties` String,
    `group4_properties` String,
    `person_created_at` DateTime64(3),
    `group0_created_at` DateTime64(3),
    `group1_created_at` DateTime64(3),
    `group2_created_at` DateTime64(3),
    `group3_created_at` DateTime64(3),
    `group4_created_at` DateTime64(3),
    `inserted_at` Nullable(DateTime64(6, 'UTC')) DEFAULT NULL,
    `person_mode` Enum8('full' = 0, 'propertyless' = 1, 'force_upgrade' = 2)
)
ENGINE = Distributed('posthog', 'posthog', 'sharded_events', sipHash64(distinct_id))
```

`writable_events` represents the schema coming in from Kafka. All materialization occurs on the `MergeTree` table.

Basically you need to do two things here:
- Create the `MATERIALIZED` column on `sharded_events`
- Add a reference to it from `events`...something like 
```
`mat_ec_<column_name>` String COMMENT 'column_materializer_ec::<column_name>',
```

I used something similar here to how we do mat cols currently for properties, just added an `_ec` for elements chain. 

You don't need to change `writable_events`.

fuziontech commented on posthog/clickhouse/migrations/0064_materialize_elements_chain.py:
```diff
@@ -0,0 +1,37 @@
+
+
+operations = [
+    run_sql_with_exceptions(f"DROP TABLE IF EXISTS events_json_mv ON CLUSTER '{CLICKHOUSE_CLUSTER}'"),
+    run_sql_with_exceptions(f"DROP TABLE IF EXISTS kafka_events_json ON CLUSTER '{CLICKHOUSE_CLUSTER}'"),
+    migrations.RunPython(add_columns_to_required_tables),
+    run_sql_with_exceptions(KAFKA_EVENTS_TABLE_JSON_SQL()),
+    run_sql_with_exceptions(EVENTS_TABLE_JSON_MV_SQL()),
```
You don't need to do any operations with Kafka tables here.

You only need to add the column to:
- `sharded_events`
- `events`

No dropping of `kafka_` tables, no dropping of `_mv` tables.

fuziontech commented on posthog/models/event/sql.py:
```diff
@@ -22,6 +22,7 @@
     lambda: f"TRUNCATE TABLE IF EXISTS {EVENTS_DATA_TABLE()} ON CLUSTER '{settings.CLICKHOUSE_CLUSTER}'"
 )
 DROP_EVENTS_TABLE_SQL = lambda: f"DROP TABLE IF EXISTS {EVENTS_DATA_TABLE()} ON CLUSTER '{settings.CLICKHOUSE_CLUSTER}'"
+DROP_DISTRIBUTED_EVENTS_TABLE_SQL = f"DROP TABLE IF EXISTS events ON CLUSTER '{settings.CLICKHOUSE_CLUSTER}'"
```
Drop this line - no need to drop `events` distributed table

timgl replied:
This is actually for the tests not production.

fuziontech replied:
oops! yeah that makes sense :saluting_face:

fuziontech reviewed (changes requested):
The logic looks solid here, but the way you are adding the materialized column is totally different from how we've done it in the past. I'd like to have this consistent since the way you are doing it here is actually more steps and will make the materialized schemas inconsistent.

thmsobrmlr committed:
refactor(insights): hidden legend indexes (#23038)

webjunkie committed:
feat(queries): Add team to async query metrics (#23149)

mariusandra committed:
feat(hog): refactor code editors, hog validation (#23054)

thmsobrmlr committed:
refactor(insights): query based insight usage reporting (#23084)

pauldambra committed:
feat: remove exceptions to ignore from decide (#23177)

thmsobrmlr committed:
refactor(insights): remove `include_query_insights` flag (#23091)

Gilbert09 committed:
feat(data-warehouse): Added incremental syncs for postgres (#23145)

* Added incremental syncs for postgres

* Tests

* Fixed mypy

* Added extra protection on the sync form

* Set the correct write disposition for SQL

* PR fixes

* Fixed tests

* Fixed tests

* Fixed tests

EDsCODE committed:
chore(data-warehouse): fix column typing and remove unnecessary UI (#23130)

* fix column typing and remove unneessary UI

* typing

Twixes committed:
fix(api): Always allow GET /users/@me with personal API key (#23188)

* fix(api): Always allow GET /users/@me with personal API key

* Fix `scope_object` retrieval

robbie-c committed:
fix(web-analytics): Make channel type comparison case-insensitive (#23182)

* Make channel type comparison case-insensitive

* Add comment

Twixes committed:
chore(insights): Yeet the unbaked groups-on-events (#22764)

* chore(insights): Yeet the unbaked groups-on-events

* Update TestAutoProjectMiddleware

* Update verifiedDomainsLogic.test.ts.snap

* Update funnel_correlation.py

* Update test_property.py

* Update query snapshots

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update query snapshots

* Update query snapshots

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

webjunkie committed:
chore: Update pyproject.toml for Python 3.11 (#23181)

tomasfarias committed:
refactor: Support batch export models as views (#23052)

* refactor: Update metrics to fetch counts at request time

* fix: Move import to method

* fix: Add function

* feat: Custom schemas for batch exports

* feat: Frontend support for model field

* fix: Clean-up

* fix: Add missing migration

* fix: Make new field nullable

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* fix: Bump migration number

* fix: Bump migration number

* refactor: Update metrics to fetch counts at request time

* fix: Actually use include and exclude events

* refactor: Switch to counting runs

* refactor: Support batch export models as views

* fix: Merge conflict

* fix: Quality check fixes

* refactor: Update metrics to fetch counts at request time

* fix: Move import to method

* fix: Add function

* fix: Typing fixes

* feat: Custom schemas for batch exports

* feat: Frontend support for model field

* fix: Clean-up

* fix: Add missing migration

* fix: Make new field nullable

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* fix: Bump migration number

* fix: Clean-up unused code

* chore: Clean-up unused function and tests

* fix: Clean-up unused function

* fix: HTTP Batch export default fields

* fix: Remove test case on new column not present in base table

* chore: Clean-up unused functions and queries

* fix: Only run extra clickhouse queries in batch exports tests

* refactor: Remove coalesce and use only inserted_at in queries

At this point, inserted_at should always be set for all batch
exports. Only historical exports require _timestamp, but backfills
have already been switched over to query based on timestamp, so they
also do not need to check for inserted_at/_timestamp.

Removing the colaesce and using only inserted_at reduces the size
of the data CH has to fetch by half.

* fix: Remove deprecated test

* fix: Add person_id to person model and enforce ordering

* refactor: Also add version column

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

marandaneto committed:
chore: add ios session replay onboarding (#23190)

webjunkie committed:
fix(insights): Fix render column tzlabel regex (#23144)

Twixes committed:
chore(schema): Remove Python codegen hack (#23189)

* chore(schema): Remove Python codegen hack

* Make StrEnum hack cross-platform

Twixes committed:
feat(events): Expose "Open as new insight" button (#23122)

* feat(events): Expose "Open as new insight" button

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

webjunkie committed:
fix: Fix FieldAliasType.resolve_database_field not implemented (#23186)

raquelmsmith committed:
fix: handle redirect on org delete properly (#22733)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

aspicer committed:
fix: saml auth upgrade (#23197)


Co-authored-by: Zach Waterfield <zlwaterfield@gmail.com>

aspicer committed:
revert: 	"chore: upgrade python to 3.11 🐍" (#23198)



Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

dmarticus committed:
feat(surveys): add configurable delay to popup surveys (#22780)

* initial commit

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* fixing the ui a bit

* write tests for the survey_popup_delay api

* remove todos that were todone

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update frontend/src/scenes/surveys/SurveyCustomization.tsx

Co-authored-by: Phani Raj <phani@posthog.com>

* Update frontend/src/types.ts

Co-authored-by: Phani Raj <phani@posthog.com>

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* code review feedback

* add hour max to survey delay

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* not sure how this was reverted, but it's back now

* was this it?

* was this it?

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update query snapshots

* Update query snapshots

* Update query snapshots

* use latest version of posthog-js to activate the survey popup delay features

* forgot to commit the lockfile

* Update query snapshots

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update query snapshots

* Update query snapshots

* Update query snapshots

* Update query snapshots

* Update query snapshots

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update query snapshots

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Phani Raj <phani@posthog.com>

EDsCODE committed:
fix(spacing): add spacing to table descrip (#23214)

* use p tag

* Update UI snapshots for `chromium` (1)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

daibhin committed:
fix: notebook migration json parsing (#23050)

nikitaevg committed:
fix(insights): support editing insights by updating source (#23092)

tomasfarias committed:
feat: Support multiple models in S3 batch export (#23105)

* refactor: Update metrics to fetch counts at request time

* fix: Move import to method

* fix: Add function

* feat: Custom schemas for batch exports

* feat: Frontend support for model field

* fix: Clean-up

* fix: Add missing migration

* fix: Make new field nullable

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* fix: Bump migration number

* fix: Bump migration number

* refactor: Update metrics to fetch counts at request time

* refactor: Switch to counting runs

* refactor: Support batch export models as views

* fix: Merge conflict

* fix: Quality check fixes

* refactor: Update metrics to fetch counts at request time

* fix: Move import to method

* fix: Add function

* fix: Typing fixes

* feat: Custom schemas for batch exports

* feat: Frontend support for model field

* fix: Clean-up

* fix: Add missing migration

* fix: Make new field nullable

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* fix: Bump migration number

* fix: Clean-up unused code

* fix: Clean-up unused function

* fix: Only run extra clickhouse queries in batch exports tests

* feat: Support multiple models in all batch export destinations

This commit introduces a new `batch_export_model` input for all batch
export destinations. This input is of type `BatchExportModel` which
can be used to indicate which model a batch export is supposed to
target. This opens the door to creating "persons" model exports, and
eventually extending this further with more changes.

The change was done in a backwards compatible way, so
`batch_export_schema` was not removed, and existing export
configurations should continue to work (as evidenced by unit tests
passing without changes).

However, moving forward all batch exports will be created with
`batch_export_schema` set to `None` and `batch_export_model` defined
in its place.

After updating existing exports, `batch_export_schema` and any other code
associated with backwards compatibility can be deleted.

* fix: Add additional type hints and update tests

* chore: Add test utilities for multi-model batch exports

* fix: Pass properties included as keyword arguments

* fix: Support custom key prefixes for multi-model

* feat: Unit testing for S3 batch exports with multi-model support

* fix: Add and correct type hints

* fix: Typo in parameter

* fix: API tests now work with model parameter

* fix: Re-add hosts to docker-compose

* fix: Use UTC timezone alias

* fix: Add missing test column

* revert: Python 3.11 alias

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

thmsobrmlr committed:
refactor(insights): query based `allEventNames` selector (#23202)

thmsobrmlr committed:
refactor(insights): query based `intervalUnit` (#23203)

webjunkie committed:
fix(exports): Fix export current columns by using person id column (#23191)

thmsobrmlr committed:
refactor(insights): remove more filter based selectors (#23194)

Gilbert09 committed:
chore(data-warehouse): Allow syncing to a single S3 folder for the life of a schema (#23155)

Allow syncing to a single S3 folder for the life of a schema

webjunkie committed:
feat(queries): Add async queries backend chaining for dashboards (#23125)

posthog-bot committed:
chore(deps): Update posthog-js to 1.141.1 (#23230)

robbie-c committed:
feat(web-analytics): Add Sessions Table V2 (#23023)

* Add raw_sessions table

* Fix

* Change time chunking to 5 minutes

* Add modes of operation, and some comments

* WIP wire up sessions table V2

* More working v2 sessions tests

* Optimize imports

* Fix v1 sessions table test

* Fix more tests

* Fix channel type tests

* Fix session replay joining with v2

* Web analytics queries and their tests working

* Fix where clause extractor tests for v1

* Fix backfill script

* Show last select query instead of first

* Run ruff

* Fix ids in tests

* Fix database init

* spelling

* Fix test_query

* Formatting

* Handle session properties with v2 session table

* Add more columns, fix some properties

* Add new properties to taxonomy

* Fix trends tests

* Fix modifiers

* Set v1 sessions table to default

* Capture viewport size

* Fix keyword arg rename

* Update query snapshots

* Update query snapshots

* Fix test_utils

* Fix test_trends

* Make it easier to run test_parser_cpp from pytcharm

* Update query snapshots

* Update query snapshots

* Add last external click url to the sessions MV

* Update query snapshots

* Add test_last_external_click_url

* Add ingest from date

* Run schema build after a rebase

* Tweak test_all

* Update query snapshots

* Run schema after rebase

* Update query snapshots

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Change ingestion date and add explaining comment

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

webjunkie committed:
fix(exports): Make exports always calculate (#23227)

daibhin committed:
fix: temporary realtime warning page (#23219)

webjunkie committed:
fix(insights): Fix async query errors not surfacing (#23233)

webjunkie committed:
fix(hogql): Fix missing import (#23224)

Fix missing import

aspicer committed:
feat: block robots from seeing shared dashboards (#23210)

robbie-c committed:
fix: Fix bad merge (#23236)

Fix bad merge

mariusandra committed:
fix(hog): C++ parser exception (#23232)

Gilbert09 committed:
feat(data-warehouse): Added a properties field to all warehouse tables (#23228)

* Added a properties field to all warehouse tables

* Fixed tests

EDsCODE committed:
chore(data-warehouse): increase limit (#23234)

* increase limit

* Update query snapshots

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

aspicer committed:
feat(performance): query improvements for trends (load less people) (#23135)

mariusandra committed:
feat(hog): hog template string editor (#23183)

dependabot[bot] committed:
chore(deps): bump scikit-learn from 1.4.0 to 1.5.0 (#23035)

posthog-bot committed:
chore(deps): Update posthog-js to 1.141.3 (#23237)

pauldambra committed:
feat: update for web vitals (#23220)

posthog-bot committed:
chore(deps): Update posthog-js to 1.141.4 (#23252)

webjunkie committed:
fix(insights): Prevent requesting refreshes of shared dashboards and insights (#23249)

zlwaterfield committed:
fix: put the xmlsec version back to 1.3.13 (#23256)

Put the xmlsec version back to 1.3.13

daibhin committed:
fix: creating surveys improvements (#23254)

aspicer committed:
feat: update loading bar to pull in queries from query log (#23244)

EDsCODE committed:
chore(data-warehouse): restore billing logic (#23205)

* restore billing logic

* Update query snapshots

* Update query snapshots

* add row limit based on billing

* remove

* restore quota limiting

* Update query snapshots

* Update query snapshots

* Update query snapshots

* Update query snapshots

* Update query snapshots

* cache key

* Update query snapshots

* Update query snapshots

* Update query snapshots

* fix test

* add old limit logic back

* Update query snapshots

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

pauldambra committed:
fix: switch default sorting column (#23258)

Co-authored-by: David Newell <d.newell1@outlook.com>

EDsCODE committed:
chore(data-warehouse): adjust time (#23267)

adjust

aspicer committed:
feat: shorten async query exponential backoff (#23245)

EDsCODE committed:
fix(data-warehouse): wrong operator (#23270)

flip operator

tomasfarias committed:
fix: Use DISTINCT ON for batch exports query (#23259)

* fix: Use DISTINCT ON for batch exports query

When using GROUP BY clickhouse doesn't yield batches of
max_block_size, so this can cause us to exceed Arrow's String type size.

* fix: Also do migrations on cluster

* Update query snapshots

* Update query snapshots

* feat: Add migration to recreate views

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

robbie-c committed:
feat(web-analytics): Add view with combined source/medium/campaign (#23265)

* Add view with combined source/medium/campaign

* Update query snapshots

* Update query snapshots

* Update query snapshots

* Update query snapshots

* Update query snapshots

* Update query snapshots

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

timgl committed:
fix

timgl committed:
Merge branch 'master' into elements-chain-materialization

timgl committed:
Merge branch 'master' into elements-chain-materialization

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

timgl committed:
fix tests

timgl committed:
fix

timgl committed:
Merge branch 'master' into elements-chain-materialization

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

fuziontech commented on posthog/clickhouse/migrations/0064_materialize_elements_chain.py:
```diff
@@ -0,0 +1,33 @@
+
+ADD_COLUMNS_EVENTS = """
+ALTER TABLE {table} ON CLUSTER {cluster}
+ADD COLUMN IF NOT EXISTS mat_ec_href String COMMENT 'column_materializer::elements_chain::href',
+ADD COLUMN IF NOT EXISTS mat_ec_texts Array(String) COMMENT 'column_materializer::elements_chain::texts',
+ADD COLUMN IF NOT EXISTS mat_ec_ids Array(String) COMMENT 'column_materializer::elements_chain::ids',
+ADD COLUMN IF NOT EXISTS mat_ec_elements Array(Enum('a', 'button', 'form', 'input', 'select', 'textarea', 'label')) COMMENT 'column_materializer::elements_chain::elements'
```
nice :sunglasses:

fuziontech commented on posthog/clickhouse/migrations/test/test_0064.py:
```diff
@@ -0,0 +1,69 @@
+        )
+
+
+class Test0064(ClickhouseDestroyTablesMixin):
```
`ClickhouseDestroyTablesMixin` :yum:

fuziontech reviewed (approved):
Looks great!
Ship it :ship:

timgl committed:
change column names
